### PR TITLE
DOCSUP-9860: document refactored storage dictionary

### PR DESCRIPTION
**Pull request by lehasm**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- Documentation for #23436

### CODE REVIEW

**Comment by lehasm** [comment]
Hello.
Noticed that exception messages for dictionary queries contain _table_ , not _dictionary_ . Although it is not directly related to the documentation.
```DETACH DICTIONARY alexey_sm.d0 PERMANENTLY;
Received exception from server (version 21.9.1):
Code: 60. DB::Exception: Received from localhost:9000. DB::Exception: Table `alexey_sm`.`d0` doesn't exist. (UNKNOWN_TABLE)
```

File: docs/en/sql-reference/statements/detach.md [hunk 1]
```diff
@@ -5,17 +5,18 @@ toc_title: DETACH
 
 # DETACH Statement {#detach}
 
-Makes the server "forget" about the existence of the table or materialized view.
+Makes the server "forget" about the existence of a table, a materialized view or a dictionary.
```
Comment (olgarev) [inline docs/en/sql-reference/statements/detach.md:8]:
```suggestion
Makes the server "forget" about the existence of a table, a materialized view, or a dictionary.
```

File: docs/en/sql-reference/statements/detach.md [hunk 2]
```diff
@@ -5,17 +5,18 @@ toc_title: DETACH
...
-Syntax:
+**Syntax**
 
 ``` sql
-DETACH TABLE|VIEW [IF EXISTS] [db.]name [ON CLUSTER cluster] [PERMANENTLY]
+DETACH TABLE|VIEW|DICTIONARY [IF EXISTS] [db.]name [ON CLUSTER cluster] [PERMANENTLY]
 ```
 
-Detaching does not delete the data or metadata for the table or materialized view. If the table or view was not detached `PERMANENTLY`, on the next server launch the server will read the metadata and recall the table/view again. If the table or view was detached `PERMANENTLY`, there will be no automatic recall.
+Detaching does not delete the data or metadata of the table, the materialized view or the dictionary. If the entity was not detached `PERMANENTLY`, on the next server launch the server will read the metadata and recall the table/view/dictionary again. If the entity was detached `PERMANENTLY`, there will be no automatic recall.
```
Comment (olgarev) [inline docs/en/sql-reference/statements/detach.md:16]:
```suggestion
Detaching does not delete the data or metadata of a table, a materialized view or a dictionary. If an entity was not detached `PERMANENTLY`, on the next server launch the server will read the metadata and recall the table/view/dictionary again. If an entity was detached `PERMANENTLY`, there will be no automatic recall.
```

File: docs/en/sql-reference/statements/exchange.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+---
+toc_priority: 49
+toc_title: EXCHANGE
+---
+
+# EXCHANGE Statement {#exchange}
+
+Exchanges the names of two tables or dictionaries atomically.
+This task can also be accomplished with a [RENAME](./rename.md) query using a temporary name. But the operation in not atomic in that case.
```
Comment (olgarev) [inline docs/en/sql-reference/statements/exchange.md:9]:
```suggestion
This task can also be accomplished with a [RENAME](./rename.md) query using a temporary name, but the operation is not atomic in that case.
```

File: docs/en/sql-reference/statements/rename.md [hunk 4]
```diff
@@ -5,18 +5,57 @@ toc_title: RENAME
 
 # RENAME Statement {#misc_operations-rename}
 
-## RENAME DATABASE {#misc_operations-rename_database}
-Renames database, it is supported only for Atomic database engine.
+Renames databases, tables or dictionaries. Several entities can be renamed in a single query.
```
Comment (olgarev) [inline docs/en/sql-reference/statements/rename.md:8]:
```suggestion
Renames databases, tables, or dictionaries. Several entities can be renamed in a single query.
```

File: docs/en/sql-reference/statements/rename.md [hunk 5]
```diff
@@ -5,18 +5,57 @@ toc_title: RENAME
...
+Note that the `RENAME` query with several entities is non-atomic operation. To swap entity names atomically, use [EXCHANGE](./exchange.md) statement.
```
Comment (olgarev) [inline docs/en/sql-reference/statements/rename.md:9]:
```suggestion
Note that the `RENAME` query with several entities is non-atomic operation. To swap entities names atomically, use [EXCHANGE](./exchange.md) statement.
```

File: docs/en/sql-reference/statements/attach.md [hunk 6]
```diff
@@ -22,10 +29,12 @@ This query is used when starting the server. The server stores table metadata as
 
 If the table was detached permanently, it won't be reattached at the server start, so you need to use `ATTACH` query explicitly.
 
-### Сreate New Table And Attach Data {#create-new-table-and-attach-data}
+## Create New Table And Attach Data {#create-new-table-and-attach-data}
 
 **With specify path to table data**
```
Comment (olgarev) [inline docs/en/sql-reference/statements/attach.md:34]:
```suggestion
### With specify path to table data {#attach-with-specified-path}
```

File: docs/en/engines/database-engines/atomic.md [hunk 7]
```diff
@@ -25,16 +25,16 @@ CREATE TABLE name UUID '28f1c61c-2970-457a-bffe-454156ddcfef' (n UInt64) ENGINE
 ```
 ### RENAME TABLE {#rename-table}
 
-`RENAME` queries are performed without changing UUID and moving table data. These queries do not wait for the completion of queries using the table and will be executed instantly.
+[RENAME](../../sql-reference/statements/rename.md) queries are performed without changing UUID and moving table data. These queries do not wait for the completion of queries using the table and will be executed instantly.
 
 ### DROP/DETACH TABLE {#drop-detach-table}
 
 On `DROP TABLE` no data is removed, database `Atomic` just marks table as dropped by moving metadata to `/clickhouse_path/metadata_dropped/` and notifies background thread. Delay before final table data deletion is specify by [database_atomic_delay_before_drop_table_sec](../../operations/server-configuration-parameters/settings.md#database_atomic_delay_before_drop_table_sec) setting.
```
Comment (olgarev) [inline docs/en/engines/database-engines/atomic.md:32]:
```suggestion
On `DROP TABLE` no data is removed, database `Atomic` just marks table as dropped by moving metadata to `/clickhouse_path/metadata_dropped/` and notifies background thread. Delay before final table data deletion is specified by the [database_atomic_delay_before_drop_table_sec](../../operations/server-configuration-parameters/settings.md#database_atomic_delay_before_drop_table_sec) setting.
```

File: docs/ru/engines/database-engines/atomic.md [hunk 8]
```diff
@@ -25,16 +25,16 @@ CREATE TABLE name UUID '28f1c61c-2970-457a-bffe-454156ddcfef' (n UInt64) ENGINE
 ```
 ### RENAME TABLE {#rename-table}
 
-Запросы `RENAME` выполняются без изменения UUID и перемещения табличных данных. Эти запросы не ожидают завершения использующих таблицу запросов и будут выполнены мгновенно.
+Запросы [RENAME](../../sql-reference/statements/rename.md) выполняются без изменения UUID и перемещения табличных данных. Эти запросы не ожидают завершения использующих таблицу запросов и будут выполнены мгновенно.
```
Comment (olgarev) [inline docs/ru/engines/database-engines/atomic.md:28]:
```suggestion
Запросы [RENAME](../../sql-reference/statements/rename.md) выполняются без изменения UUID и перемещения табличных данных. Эти запросы не ожидают завершения использующих таблицу запросов и выполняются мгновенно.
```

File: docs/en/engines/database-engines/atomic.md [hunk 9]
```diff
@@ -25,16 +25,16 @@ CREATE TABLE name UUID '28f1c61c-2970-457a-bffe-454156ddcfef' (n UInt64) ENGINE
 ```
 ### RENAME TABLE {#rename-table}
 
-`RENAME` queries are performed without changing UUID and moving table data. These queries do not wait for the completion of queries using the table and will be executed instantly.
+[RENAME](../../sql-reference/statements/rename.md) queries are performed without changing UUID and moving table data. These queries do not wait for the completion of queries using the table and will be executed instantly.
```
Comment (olgarev) [inline docs/en/engines/database-engines/atomic.md:28]:
```suggestion
[RENAME](../../sql-reference/statements/rename.md) queries are performed without changing UUID and moving table data. These queries do not wait for the completion of queries using the table and are executed instantly.
```

File: docs/ru/sql-reference/statements/attach.md [hunk 10]
```diff
@@ -3,16 +3,23 @@ toc_priority: 40
 toc_title: ATTACH
 ---
 
-# ATTACH Statement {#attach}
+# Выражение ATTACH {#attach}
```
Comment (olgarev) [inline docs/ru/sql-reference/statements/attach.md:6]:
```suggestion
# ATTACH {#attach}
```

File: docs/ru/sql-reference/statements/exchange.md [hunk 11]
```diff
@@ -0,0 +1,42 @@
+---
+toc_priority: 49
+toc_title: EXCHANGE
+---
+
+# Выражение EXCHANGE {#exchange}
```
Comment (olgarev) [inline docs/ru/sql-reference/statements/exchange.md:6]:
```suggestion
# EXCHANGE {#exchange}
```

File: docs/ru/sql-reference/statements/rename.md [hunk 12]
```diff
@@ -3,20 +3,58 @@ toc_priority: 48
 toc_title: RENAME
 ---
 
-# RENAME Statement {#misc_operations-rename}
+# Выражение RENAME {#misc_operations-rename}
```
Comment (olgarev) [inline docs/ru/sql-reference/statements/rename.md:6]:
```suggestion
# RENAME {#misc_operations-rename}
```

**Review by kitaisreal — APPROVED** [review]

**Pull request merged**